opendrive2lanelet2/convertor/opendrive2lanelet2convertor.py

```diff
@@ -110,51 +140,43 @@ def process_vertices(self, vertices, relation_id, bound_id):
         for j in range(len(vertices)):
             node_id = relation_id + str(bound_id) + "{0:0=3d}".format(j)
             node = self.convert_vertice_to_node(node_id, vertices[j])
+            node_latlon = np.array([node.lat, node.lon])
 
-            duplicate_flag = False
-            for i in range(len(self.all_nodes)):
-                if(self.calc_distance_node_tresh(node,self.all_nodes[i])):
-                    nodes.append(self.all_nodes[i])
-                    duplicate_flag = True
-                    break
-
-                if(self.calc_distance_node_isclose(node,self.all_nodes[i])):
-                    nodes.append(self.all_nodes[i])
-                    duplicate_flag = True
-                    break
-
-            if(duplicate_flag==False):
+            is_dup, index = self.check_duplicate_hash_precise(node_latlon)
+            if is_dup:
+                nodes.append(self.all_nodes[index])
+            else:
                 nodes.append(node)
                 self.nodes.append(node.create_xml_node_object())
                 self.all_nodes.append(node)
+
         return nodes
 
     # check for way duplication in the entire map
     def check_way_duplication(self,nodes,way):
-	# For a duplicate, need at least 5, and at least 80% matching points
+    # For a duplicate, need at least 5, and at least 80% matching points
```

Comment:
5 what? Points? Please specify in the comment

Reply:
I added some comments to describe this better